core/src/main/java/hudson/model/Run.java

```diff
@@ -1538,7 +1539,19 @@ public String toString() {
      * @since 1.349
      */
     public void writeLogTo(long offset, @NonNull XMLOutput out) throws IOException {
-        getLogText().writeHtmlTo(offset, out.asWriter());
+        long start = offset;
+        if (offset > 0) {
+            try (BufferedInputStream bufferedInputStream = new BufferedInputStream(getLogInputStream())) {
+                if (offset == bufferedInputStream.skip(offset)) {
+                    int r;
+                    do {
+                        r = bufferedInputStream.read();
+                        start = (r == -1)? 0 : start + 1;
```

Comment:
So it will print the entire log if we hit the end of the stream, which defeats purpose of `offset` for large logs. Should not be a problem for real use-cases, but we might want to retain original behavior.

```suggestion
                        start = (r == -1)? 0 : offset + 1;
```

Reply:
The code change is not clear to me. My idea was to advance the `start` position for each byte that is different than `\n`. What this change would do is re-set `start` to the value of `offset + 1` multiple times - `offset` doesn't get updated anywhere. Can you please explain what you had in mind?

This immediately breaks 2 tests which I think are valid use cases.